translate/translate.js

```diff
@@ -18,156 +18,128 @@
-  });
+  return translate.getLanguages(target)
+    .then((results) => {
+      const languages = results[0];
+
+      console.log('Languages:');
+      languages.forEach((language) => console.log(language));
+      return languages;
+    });
 }
 // [END translate_list_language_names]
 
 // [START translate_translate_text]
-function translateText (input, target, callback) {
+function translateText (input, target) {
+  if (!Array.isArray(input)) {
+    input = [input];
+  }
+
   // Instantiates a client
-  const translate = Translate({
-    // The Translate API uses an API key for authentication. This sample looks
-    // for the key in an environment variable.
-    key: process.env.TRANSLATE_API_KEY
-  });
+  const translate = Translate();
 
   // Translates the text into the target language. "input" can be a string for
   // translating a single piece of text, or an array of strings for translating
   // multiple texts.
-  translate.translate(input, target, (err, translation) => {
-    if (err) {
-      callback(err);
-      return;
-    }
-
-    console.log('Text: %j', input);
-    console.log('Translation: %j', translation);
-    callback();
-  });
+  return translate.translate(input, target)
+    .then((results) => {
+      let translations = results[0];
+      translations = Array.isArray(translations) ? translations : [translations];
+
+      console.log('Translations:');
+      translations.forEach((translation, i) => {
+        console.log(`${input[i]} => (${target}) ${translation}`);
+      });
+
+      return translations;
+    });
 }
 // [END translate_translate_text]
 
-// The command-line program
-const cli = require(`yargs`);
-const noop = require(`../utils`).noop;
-
-const program = module.exports = {
-  detectLanguage: detectLanguage,
-  listLanguages: listLanguages,
-  listLanguagesWithTarget: listLanguagesWithTarget,
-  translateText: translateText,
-  main: (args) => {
-    // Run the command-line program
-    cli.help().strict().parse(args).argv;
-  }
-};
-
-cli
+require(`yargs`)
   .demand(1)
-  .command(`detect <input..>`, `Detects the language of the provided text or texts`, {}, (opts) => {
-    if (!process.env.TRANSLATE_API_KEY) {
-      process.env.TRANSLATE_API_KEY = opts.apiKey;
-    }
-    program.detectLanguage(opts.input, noop);
-  })
-  .command(`list [target]`, `Lists available translation languages. To return language names in a language other than English, specify a target language.`, {}, (opts) => {
-    if (!process.env.TRANSLATE_API_KEY) {
-      process.env.TRANSLATE_API_KEY = opts.apiKey;
+  .command(
+    `detect <input..>`,
+    `Detects the language of one or more strings.`,
+    {},
+    (opts) => detectLanguage(opts.input)
+  )
+  .command(
+    `list [target]`,
+    `Lists available translation languages. To return language names in a language other than English, specify a target language.`,
+    {},
+    (opts) => {
+      if (opts.target) {
+        listLanguagesWithTarget(opts.target);
```

Comment:
Wishful thinking: `listLanguages` goes away, and `listLanguagesWithTarget` just uses `en` as a default.